Makefile

```diff
@@ -0,0 +1,53 @@
+DEFAULT_GOAL := help
+
+# Include .env and export it so variables set in there are available in the
+# Makefile.
+include .env
+export
+
+# Set these in the environment to override them. This is helpful for
+# development if you have file ownership problems because the user in the
+# container doesn't match the user on your host.
+USER_ID ?= 10001
+GROUP_ID ?= 10001
+
+.PHONY: help
+help:
+	$(info Available rules:)
+	$(info )
+	@fgrep -h "##" Makefile | fgrep -v fgrep | sed 's/\(.*\):.*##/\1:/' | column -t -s '|'
+
+.docker-build:
+	make build
+
+.env:
+	@if [ ! -f .env ]; \
+	then \
+	echo "Creating .env file..."; \
+	echo "# USER_ID=\n# GROUP_ID=\n" > .env; \
+	fi
```

Comment:
When running black in the container on macOS, everything is fine because docker on macOS is "fun" about what user it's running as and permissions to the file system.

On Linux, this fails because the container has a uid/gid of 10001/10001 and that's not the same as my uid/gid, so then when I run black, I just get a lot of permission issues. So I do this thing where it allows for a `.env` override of what uid/gid to use to build the user in the container so then I don't get permission issues, but it does it in a way that doesn't affect anyone else and works fine on macOS.

Reply:
Nice! I knew about that problem but didn't know about that solution

Reply:
It's the least painful one I've found so far. I'd love to know if there are better ones.

I tossed around tweaking it to auto-detect whether the host is macos or linux and set these accordingly. I'll tinker with that some day.